### PR TITLE
LMW-530

### PR DESCRIPTION
**Pull request by whoislyuboanyway**

### CODE REVIEW

File: WalletCore/Classes/Core/Model/Assets/Asset.swift [hunk 1]
```diff
@@ -14,7 +14,7 @@ public class Asset {
     public var percent: Double
```
Comment (whoislyuboanyway) [inline WalletCore/Classes/Core/Model/Assets/Asset.swift:14]: Shouldn't this be `Decimal`?

**Review by vladimirdimovprimeholding — APPROVED** [review]

File: WalletCore/Classes/ViewModel/TransactionsHeaderViewModel.swift [hunk 2]
```diff
@@ -81,28 +82,27 @@ fileprivate extension ObservableType where Self.E == ApiResult<LWAssetModel> {
...
-                accuracy: Int(data.asset.accuracy)
-            )}
-            .map{"(\($0))"}
-            .startWith("")
+fileprivate extension ObservableType where Self.E == (wallets: [LWSpotWallet], asset: LWAssetModel) {
+    func mapToAssetBalance() -> Observable<String> {
+        return map { value in
+            return value.wallets.filter { $0.asset.identity == value.asset.identity }
+                .map { $0.amountInBase.decimalValue }
+                .reduce(0.0, +)
```
Comment (StanevPrime) [inline WalletCore/Classes/ViewModel/TransactionsHeaderViewModel.swift:90]: Implement an extension for calculating balance

File: WalletCore/Classes/ViewModel/WalletViewModel.swift [hunk 3]
```diff
@@ -66,9 +66,20 @@ public class WalletViewModel {
             .asDriver(onErrorJustReturn: "")
             .startWith("")
         
-        percent = Observable.combineLatest(wallet, mainInfoObservable) { (assetAmount: $0.amountInBase.doubleValue, totalAmount: $1.mainInfo.totalBalance.doubleValue) }
+        let totalAmountObservable = nonEmptyWallets
+            .map { $0.map { $0.amountInBase.decimalValue }.reduce(0.0, +) }
```
Comment (StanevPrime) [inline WalletCore/Classes/ViewModel/WalletViewModel.swift:70]: Implement an extension for calculating balance

File: WalletCore/Classes/ViewModel/WalletsViewModel.swift [hunk 4]
```diff
@@ -10,43 +10,89 @@ import Foundation
...
-            .map{ mainInfo, wallets, assets in
-                (asset: mainInfo.asset, wallets: wallets, mainInfo: mainInfo.mainInfo, assets: assets)
+        let infoObservable = Observable<InfoData>
+            .combineLatest(baseAsset.filterSuccess(), nonEmptyWallets.filterSuccess(), allAssets.filterSuccess())
+            {
+                (
+                    asset: $0,
+                    wallets: $1,
+                    assets: $2,
+                    totalBalance: $1.map { $0.amountInBase.decimalValue }.reduce(0.0, +)
```
Comment (StanevPrime) [inline WalletCore/Classes/ViewModel/WalletsViewModel.swift:56]: Implement an extension for calculating balance

**Review by StanevPrime — COMMENTED** [review]
Implement a extension for calculating balance

File: WalletCore/Classes/ViewModel/WalletsViewModel.swift [hunk 5]
```diff
@@ -10,43 +10,89 @@ import Foundation
 import RxSwift
 import RxCocoa
 
+public typealias InfoData = (
```
Comment (StanevPrime) [inline WalletCore/Classes/ViewModel/WalletsViewModel.swift:13]: Please use meaningful name for typealias or nest it inside the context.

**Pull request merged**